server/controllers/request.js

```diff
@@ -0,0 +1,64 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler")
+
+// @route POST /requests
+// @desc Create new request
+// @access Public
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const { requester, sitter, petIds, startDate, endDate } = req.body
+  const request = await Request.create({requester, sitter, petIds, startDate, endDate});
+
+  if (request) {
+    res.status(200).json({
+      success: {
+        request: {
+          ...request
+        }
+      }
+    })
+  }
+});
+
+// @route PUT /requests/edit/:requestId?userType=
+// @desc edit request
+// @access Public
+exports.editRequest = asyncHandler(async (req, res, next) => {
+  // Until we implement cancelling bookings, owners shouldn't be able to edit bookings
+  if (req.query.userType != "sitter") {
```

Comment:
Any user will be able to send these optional query parameters to edit this request :)
Please look at the example below to further understand how we would go about this issue:
```
const userId = req.user.id
const requestId = req.params.requestId

if (!mongoose.Types.ObjectId.isValid(requestId)) {
   return res.status(400).send("Bad Request");
 }

const request = Request.findById(requestId)

if (request.sitter === userId) {
   //This implies that the logged in user is a sitter and can edit (accept/decline) this request
} else {
   res.status(403)
   throw new Error ("You are not authorized to perform this operation")
}

```

Reply:
Ya I didn't understand how user object was getting passed in without params until Gurkirt told me last night, so I changed it back. Matching the sitter field to the user Id seems pretty obvious in retrospect. Woops